### ISSUE TITLE
CTS results: backport-1.0.13-2

### ISSUE THREAD

**Issue by kskmori** [ClusterLabs/pacemaker-1.0#12]
Versions:

  Pacemaker: https://github.com/kskmori/pacemaker-1.0/commit/41e7f50faa3a7446cad8a0a03df37abca34601e5
  heartbeat-3.0.5
  corosync-1.4.4
  cluster-glue-1.0.10
  resource-agents-3.9.3

Overall results:

(1) RHEL6 / heartbeat
  Feb 06 09:50:32   Overall Results:{'failure': 0, 'skipped': 0, 'success': 300, 'BadNews': 0}

(2) RHEL6 / corosync
  Feb 05 07:57:21   Overall Results:{'failure': 0, 'skipped': 0, 'success': 300, 'BadNews': 83}

(3) RHEL5 / heartbeat
  Feb 05 07:22:51   Overall Results:{'failure': 0, 'skipped': 0, 'success': 300, 'BadNews': 1}

(4) RHEL5 / corosync (aborted)
  Feb 05 22:33:55   Running test Restart                (cts0101)     [ 20] ,154:2
  command timed out: 1200 seconds without output, killing pid 18567

**Comment by beekhof** [comment]
Do you have more information for (2) and (4) ?

**Comment by beekhof** [comment]
I should add that this CTS testing is really helpful!

**Comment by kskmori** [comment]
Oh, you quickly noticed:). Yes, I'm going to add more. I can send the full logs if you want but it is pretty big.

**Comment by kskmori** [comment]
Ah sorry , I wanted to add an issue to my personal branch..

**Comment by beekhof** [comment]
On 07/02/2013, at 5:04 PM, Keisuke MORI notifications@github.com wrote:

> Oh, you quickly noticed:). Yes, I'm going to add more. I can send the full logs if you want but it is pretty big.

Gmail should be able to handle it :)

**Comment by kskmori** [comment]
BadNews on corosync stack (2)(4):

(1)

```
 Feb 04 22:13:23 BadNews: Feb  4 22:08:24 cts0201 crmd: [2777]: info: te_fence_node: Executing reboot fencing operation (16) on cts0202 (timeout=60000)
```

It should be disapperd with https://github.com/kskmori/pacemaker-1.0/commit/3915eaa74ef6a1403c31d8061c642b82824648bc

(2) 

```
Feb 05 01:11:21 Running test ComponentFail          (cts0202)     [ 93]
Feb 05 01:13:07 BadNews: Feb  5 01:10:39 cts0202 crmd: [2867]: ERROR: stonithd_op_result_ready: not signed on
Feb 05 01:13:08 BadNews: Feb  5 01:10:39 cts0202 lrmd: [3325]: ERROR: stonithd_virtual_stonithRA_ops: failed to fetch reply
Feb 05 01:13:08 BadNews: Feb  5 01:10:39 cts0202 lrmd: [3325]: ERROR: sending stonithRA op to stonithd failed.
```

(3) 

```
Feb 05 02:09:01 Running test ComponentFail          (cts0201)     [122]
Feb 05 02:10:33 BadNews: Feb  5 02:08:03 cts0201 crmd: [3793]: ERROR: stonithd_op_result_ready: not signed on
Feb 05 02:10:33 BadNews: Feb  5 02:08:03 cts0201 lrmd: [4250]: ERROR: execra:189: Cannot sign on the stonithd.
```

(4) 

```
Feb 05 02:34:31 Running test ComponentFail          (cts0202)     [136]
Feb 05 02:36:01 BadNews: Feb  5 02:33:32 cts0202 crmd: [1738]: ERROR: stonithd_op_result_ready: not signed on
```

(2),(3),(4) look a timing issue of stonithd component fail.
I have observed at least (4) on 1.0.12 too.

(5) CTS aborted

```
Feb 05 22:30:42 Running test Stonithd               (cts0101)     [ 18]
Feb 05 22:33:09 Running test SimulStop              (cts0101)     [ 19]
Feb 05 22:33:55 Running test Restart                (cts0101)     [ 20],154:2
command timed out: 1200 seconds without output, killing pid 18567
process killed by signal 9
program finished with exit code -1
elapsedTime=3174.064560
```

I will send the full logs later via gmail.

**Comment by kskmori** [comment]
I have observed the CTS abortion (5) in 1.0.12 too. I guess that it is a know issue of the plug-in mode about a race in threading when shutting down corosync.

I personally think that if those issues are known issues in 1.0.12 then we can leave them in 1.0.13 too as they are not regressions at least.

**Comment by kskmori** [comment]
BadNews on Heartbeat stack:

(6)

```
Feb 05 03:02:24 Running test ComponentFail          (cts0105)     [157]
Feb 05 03:04:38 BadNews: Feb  5 03:02:31 s_local@cts0105 crmd: [7685]: ERROR: attrd_connection_destroy: Lost connection to attrd
```

It's also a timing issue of attrd component fail. so I think this is harmless.

**Comment by kskmori** [comment]
> I have observed the CTS abortion (5) in 1.0.12 too. I guess that it is a know issue of the plug-in mode about a race in threading when shutting down corosync.

Sorry, I was totally wrong. It seems like a bug in the buildbot I'm using along with CTS.
http://trac.buildbot.net/ticket/2176

So please just ignore this part. Sorry for blaming it.

**Comment by beekhof** [comment]
On 07/02/2013, at 5:48 PM, Keisuke MORI notifications@github.com wrote:

> BadNews on Heartbeat stack:
> 
> (6)
> 
> Feb 05 03:02:24 Running test ComponentFail          (cts0105)     [157]
> Feb 05 03:04:38 BadNews: Feb  5 03:02:31 s_local@cts0105 crmd: [7685]: ERROR: attrd_connection_destroy: Lost connection to attrd
> 
> It's also a timing issue of attrd component fail. so I think this is harmless.

Correct. Completely harmless. The crmd is asynchronously noticing that attrd died.

**Comment by beekhof** [comment]
On 07/02/2013, at 7:58 PM, Keisuke MORI notifications@github.com wrote:

> I have observed the CTS abortion (5) in 1.0.12 too. I guess that it is a know issue of the plug-in mode about a race in threading when shutting down corosync.
> 
> Sorry, I was totally wrong. It seems like a bug in the buildbot I'm using along with CTS.
> http://trac.buildbot.net/ticket/2176
> 
> So please just ignore this part. Sorry for blaming it.

:)

You're running CTS via buildbot?

**Comment by kskmori** [comment]
Yes, actually Tomo Nozawa-san, my colleague, is running the buildbot and CTS.

**Comment by beekhof** [comment]
Maybe he can share the config?  I'd love to know how he's doing it :)

Based on everything I've seen so far, I'd vote to tag 1.0.13 immediately.  It looks very solid.

**Comment by kskmori** [comment]
Andrew Beekhof notifications@github.com writes:

> Maybe he can share the config?  I'd love to know how he's doing it :)

Sure. He wrote some scripts for that. I will ask him to send it to you.

> Based on everything I've seen so far, I'd vote to tag 1.0.13 immediately.  It looks very solid.

Great! I'd agree with it.
Could you merge the pull request #11 before tagging?
Or I can do that if you are ok.
https://github.com/ClusterLabs/pacemaker-1.0/pull/11

Thanks,

**Comment by kskmori** [comment]
Closed as 1.0.13 has been released. Thanks!
